### PR TITLE
Generates `InjectedTest` in a named package.

### PR DESCRIPTION
**Pull request by Vlatombe**
This makes the test friendlier to JUnit results collection when working with PCT or in multi-module projects in general.

The package name is derived from groupId and artifactId, but can also be provided through configuration if needed.

### Testing done

Existing IT `parent-4x` exercises this, though doesn't make any assertion about the actual package name being used. Could add a new one if needed.

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Review by basil — APPROVED** [review]
+1, seems reasonable to use a unique name for each plugin to facilitate test result visualizations.

**Review by jglick — APPROVED** [review]
Looks helpful. Should not break any tools which run `-Dtest=InjectedTest` since that will match a test inside a package.

File: src/test/java/org/jenkinsci/maven/plugins/hpi/TestInsertionMojoTest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+public class TestInsertionMojoTest {
+
+    @Test
+    public void legalizePackageName() {
+        assertEquals(
+                "org.jenkinsci.maven.plugins.hpi",
+                TestInsertionMojo.legalizePackageName("org.jenkinsci.maven.plugins.hpi"));
+        assertEquals(
+                "_123org.jenkinsci.maven.plugins.hpi",
+                TestInsertionMojo.legalizePackageName("123org.jenkinsci.maven.plugins.hpi"));
```
Comment (jglick) [inline src/test/java/org/jenkinsci/maven/plugins/hpi/TestInsertionMojoTest.java:16]: I suppose you might as well test `-` → `_` here.

**Review by jglick — COMMENTED** [review]
I would suggest checking `javax.lang.model.SourceVersion.isName(CharSequence)` on the candidate package components and if not, either
* fixing it, like prepending `_`
* reporting a clear error and asking you to override the property
* falling back to putting the test in the default package

**Review by jglick — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Pull request merged**